integration_test/package.json

```diff
@@ -5,7 +5,7 @@
   "scripts": {
     "clean": "rm -rf ts/build && rm -rf ts/build-node",
     "build:testserver": "cd go && go build -o ./build/testserver ./testserver/testserver.go",
-    "build:proto": "cd proto && prototool generate",
+    "build:proto": "cd proto && buf generate",
```

Comment:
I'm currently getting this error:
 ```
npm run build:proto

> grpc-web-integration-test@0.14.0 build:proto
> cd proto && buf generate

Unknown flag: --js_opt
plugin js: exit status 1
```

Am I missing some dependencies or likely to have an old version of something?

Running this in CI and checking if there's a diff might be a sensible step to avoid this breaking and just to ensure that the generated code is consistent (this will require locking down all of the versions to avoid latest being used anywhere and breaking unexpectedly at some point in the future).

I'm also not clear on how the version for `protoc-gen-go` that I know is being used here is meant to be specified (related to other comment).

Reply:
The error implies that you have a very old version of `protoc` installed that doesn't support the `--js_opt` flag to configure the `protoc` builtin javascript plugin. You can run the script CI uses to install the same version of protoc as is used in the CI job (`./install-protoc.sh`). `protoc` is only used for the javascript plugin.

I've added a step to CI that should regenerate and error if there was a diff, hopefully that should cover any regeneration issues.

The version of `protoc-gen-go` is now tracked in the `go.mod` file, thanks to `tools.go` and using `go install` to install it. This guarantees that the version of the Go protobuf runtime library and the generator version always match.